src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -24,163 +29,187 @@ case class CromwellRuntimeAttributes(docker: Option[String],
-
-    validatedRuntimeAttributes match {
-      case Success(r) => r
-      case Failure(f) =>
-        throw new IllegalArgumentException() with ThrowableWithErrors {
-          val message = "RuntimeAttribute is not valid."
-          val errors = f
-        }
+    unsupportedKeys(wdlRuntimeAttributes.attrs.keys, backendType) foreach log.warn
+    validateRequiredKeys(wdlRuntimeAttributes.attrs, backendType) match {
+      case scalaz.Success(x) => scala.util.Success(wdlRuntimeAttributes)
+      case scalaz.Failure(e) => scala.util.Failure(new IllegalArgumentException() with ThrowableWithErrors {
+        val message = "RuntimeAttributes are invalid."
+        val errors = e
+      })
     }
   }
 
+  def unsupportedKeys(keys: Iterable[String], backendType: BackendType): Option[String] = {
+    val supportedKeys = backendType.supportedKeys map { _.key }
+    val unsupportedKeys = keys.toSet -- supportedKeys
+
+    if (unsupportedKeys.isEmpty) None
+    else Option(s"Found unsupported keys for backend '$backendType': " + unsupportedKeys.toSeq.sorted.mkString(", "))
+  }
+
   val LocalDiskName = "local-disk"
-  val LocalizationDisk = LocalDisk(LocalDiskName, DiskType.SSD, Option(10)).toDisk
-
-  /** Fallback values if values for these keys are not specified in a task "runtime" stanza. */
-  object Defaults {
-    val Cpu = 1L
-    val Disk = Vector(LocalizationDisk)
-    val FailOnStderr = false
-    val ContinueOnReturnCode = ContinueOnReturnCodeSet(Set(0))
-    val Memory = 2.0
-    val Preemptible = 0
-    val Zones = Vector("us-central1-a")
-  }
-
-  private def validateRuntimeAttributes(attributeMap: AttributeMapTrait): ValidationNel[String, CromwellRuntimeAttributes] = {
-    val docker = attributeMap.get(DOCKER)
-    val zones = attributeMap.get(ZONES) map { _.split("\\s+").toVector } getOrElse Defaults.Zones
+  val LocalizationDisk = LocalDisk(LocalDiskName, DiskType.SSD, sizeGb=10)
+
+  private val keys = Set("cpu", "disks", "docker", "zones", "continueOnReturnCode", "failOnStderr", "preemptible", "memory")
+
+  private val defaultValues = Map(
+    "cpu" -> WdlInteger(1),
+    "disks" -> WdlString(LocalizationDisk.toString),
+    "zones" -> WdlString("us-central1-a"),
+    "continueOnReturnCode" -> WdlInteger(0),
+    "failOnStderr" -> WdlBoolean.False,
+    "preemptible" -> WdlInteger(0),
+    "memory" -> WdlString("2GB")
+  )
+
+  private def defaultValues(workflowOptions: WorkflowOptions): Map[String, WdlValue] = {
+    keys.collect({
+      case k if workflowOptions.getDefaultRuntimeOption(k).isSuccess =>
+        k -> WdlString(workflowOptions.getDefaultRuntimeOption(k).get)
+      case k if defaultValues.contains(k) =>
+        k -> defaultValues.get(k).get
+    }).toMap
+  }
+
+  val defaults: CromwellRuntimeAttributes = {
+    // .get below because we assume that the default values are valid
+    // There's a unit test in CromwellRuntimeAttributeSpec to test these defaults
+    validateRuntimeAttributes(defaultValues).get
```

Comment:
I hate raw .gets as much as anyone who'd ever look at this code, but there's a comment 2 lines above as to why it's being called this way